[Bug] NestedupdateAgg key fieldName maybe shuld be trimed

### Search before asking

- [x] I searched in the [issues](https://github.com/apache/paimon/issues) and found nothing similar.


### Paimon version

master

### Compute Engine

flink 1.17

### Minimal reproduce step

When create table nested-key with field space
`WITH (
	'bucket'='6',   
 ...
    'fields.dispath_array.aggregate-function' = 'nested_update',
    'fields.dispath_array.nested-key' = 'dispatch_id, order_serial_no',
    'fields.item_array.aggregate-function' = 'nested_update',
    'fields.item_array.nested-key' = 'item_id, order_serial_no'
);`
```java
Caused by: java.lang.RuntimeException: Could not instantiate generated class 'interface org.apache.paimon.codegen.Projection'
	at org.apache.paimon.codegen.CodeGenUtils.generate(CodeGenUtils.java:115) ~[paimon-flink-1.17-1.1-SNAPSHOT.jar:1.1-SNAPSHOT]
	at org.apache.paimon.codegen.CodeGenUtils.newProjection(CodeGenUtils.java:56) ~[paimon-flink-1.17-1.1-SNAPSHOT.jar:1.1-SNAPSHOT]
	at org.apache.paimon.codegen.CodeGenUtils.newProjection(CodeGenUtils.java:48) ~[paimon-flink-1.17-1.1-SNAPSHOT.jar:1.1-SNAPSHOT]
	at org.apache.paimon.mergetree.compact.aggregate.FieldNestedUpdateAgg.<init>(FieldNestedUpdateAgg.java:62) ~[paimon-flink-1.17-1.1-SNAPSHOT.jar:1.1-SNAPSHOT]
	at org.apache.paimon.mergetree.compact.aggregate.factory.FieldNestedUpdateAggFactory.createFieldNestedUpdateAgg(FieldNestedUpdateAggFactory.java:59) ~[paimon-flink-1.17-1.1-SNAPSHOT.jar:1.1-SNAPSHOT]
	at org.apache.paimon.mergetree.compact.aggregate.factory.FieldNestedUpdateAggFactory.create(FieldNestedUpdateAggFactory.java:40) ~[paimon-flink-1.17-1.1-SNAPSHOT.jar:1.1-SNAPSHOT]
	at org.apache.paimon.mergetree.compact.aggregate.factory.FieldAggregatorFactory.create(FieldAggregatorFactory.java:50) ~[paimon-flink-1.17-1.1-SNAPSHOT.jar:1.1-SNAPSHOT]
	at org.apache.paimon.mergetree.compact.aggregate.AggregateMergeFunction$Factory.create(AggregateMergeFunction.java:150) ~[paimon-flink-1.17-1.1-SNAPSHOT.jar:1.1-SNAPSHOT]
	at org.apache.paimon.mergetree.compact.MergeFunctionFactory.create(MergeFunctionFactory.java:30) ~[paimon-flink-1.17-1.1-SNAPSHOT.jar:1.1-SNAPSHOT]
	at org.apache.paimon.operation.KeyValueFileStoreWrite.createWriter(KeyValueFileStoreWrite.java:222) ~[paimon-flink-1.17-1.1-SNAPSHOT.jar:1.1-SNAPSHOT]
	at org.apache.paimon.operation.KeyValueFileStoreWrite.createWriter(KeyValueFileStoreWrite.java:95) ~[paimon-flink-1.17-1.1-SNAPSHOT.jar:1.1-SNAPSHOT]
	at org.apache.paimon.operation.AbstractFileStoreWrite.createWriterContainer(AbstractFileStoreWrite.java:445) ~[paimon-flink-1.17-1.1-SNAPSHOT.jar:1.1-SNAPSHOT]
	at org.apache.paimon.operation.AbstractFileStoreWrite.lambda$getWriterWrapper$5(AbstractFileStoreWrite.java:407) ~[paimon-flink-1.17-1.1-SNAPSHOT.jar:1.1-SNAPSHOT]
	at java.util.HashMap.computeIfAbsent(HashMap.java:1127) ~[na:1.8.0_251]
	at org.apache.paimon.operation.AbstractFileStoreWrite.getWriterWrapper(AbstractFileStoreWrite.java:406) ~[paimon-flink-1.17-1.1-SNAPSHOT.jar:1.1-SNAPSHOT]
	at org.apache.paimon.operation.AbstractFileStoreWrite.write(AbstractFileStoreWrite.java:157) ~[paimon-flink-1.17-1.1-SNAPSHOT.jar:1.1-SNAPSHOT]
	at org.apache.paimon.table.sink.TableWriteImpl.writeAndReturn(TableWriteImpl.java:175) ~[paimon-flink-1.17-1.1-SNAPSHOT.jar:1.1-SNAPSHOT]
	at org.apache.paimon.flink.sink.StoreSinkWriteImpl.write(StoreSinkWriteImpl.java:189) ~[paimon-flink-1.17-1.1-SNAPSHOT.jar:1.1-SNAPSHOT]
	at org.apache.paimon.flink.sink.GlobalFullCompactionSinkWrite.write(GlobalFullCompactionSinkWrite.java:113) ~[paimon-flink-1.17-1.1-SNAPSHOT.jar:1.1-SNAPSHOT]
	at org.apache.paimon.flink.sink.RowDataStoreWriteOperator.processElement(RowDataStoreWriteOperator.java:142) ~[paimon-flink-1.17-1.1-SNAPSHOT.jar:1.1-SNAPSHOT]
	... 13 common frames omitted
Caused by: org.apache.paimon.shade.guava30.com.google.common.util.concurrent.UncheckedExecutionException: java.lang.ArrayIndexOutOfBoundsException: -1
	at org.apache.paimon.shade.guava30.com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2051) ~[paimon-flink-1.17-1.1-SNAPSHOT.jar:1.1-SNAPSHOT]
	at org.apache.paimon.shade.guava30.com.google.common.cache.LocalCache.get(LocalCache.java:3962) ~[paimon-flink-1.17-1.1-SNAPSHOT.jar:1.1-SNAPSHOT]
	at org.apache.paimon.shade.guava30.com.google.common.cache.LocalCache$LocalManualCache.get(LocalCache.java:4859) ~[paimon-flink-1.17-1.1-SNAPSHOT.jar:1.1-SNAPSHOT]
	at org.apache.paimon.codegen.CodeGenUtils.generate(CodeGenUtils.java:110) ~[paimon-flink-1.17-1.1-SNAPSHOT.jar:1.1-SNAPSHOT]
	... 32 common frames omitted
Caused by: java.lang.ArrayIndexOutOfBoundsException: -1
	at java.util.ArrayList.elementData(ArrayList.java:422) ~[na:1.8.0_251]
	at java.util.ArrayList.get(ArrayList.java:435) ~[na:1.8.0_251]
	at java.util.Collections$UnmodifiableList.get(Collections.java:1311) ~[na:1.8.0_251]
	at java.util.stream.IntPipeline$4$1.accept(IntPipeline.java:250) ~[na:1.8.0_251]
	at java.util.Spliterators$IntArraySpliterator.forEachRemaining(Spliterators.java:1032) ~[na:1.8.0_251]
	at java.util.Spliterator$OfInt.forEachRemaining(Spliterator.java:693) ~[na:1.8.0_251]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482) ~[na:1.8.0_251]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472) ~[na:1.8.0_251]
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708) ~[na:1.8.0_251]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[na:1.8.0_251]
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499) ~[na:1.8.0_251]
	at org.apache.paimon.utils.TypeUtils.project(TypeUtils.java:79) ~[paimon-flink-1.17-1.1-SNAPSHOT.jar:1.1-SNAPSHOT]
	at org.apache.paimon.codegen.CodeGeneratorImpl.generateProjection(CodeGeneratorImpl.java:32) ~[na:na]
	at org.apache.paimon.codegen.CodeGenUtils.lambda$newProjection$1(CodeGenUtils.java:60) ~[paimon-flink-1.17-1.1-SNAPSHOT.jar:1.1-SNAPSHOT]
	at org.apache.paimon.codegen.CodeGenUtils.generateClass(CodeGenUtils.java:127) ~[paimon-flink-1.17-1.1-SNAPSHOT.jar:1.1-SNAPSHOT]
	at org.apache.paimon.codegen.CodeGenUtils.lambda$generate$5(CodeGenUtils.java:110) ~[paimon-flink-1.17-1.1-SNAPSHOT.jar:1.1-SNAPSHOT]
	at org.apache.paimon.shade.guava30.com.google.common.cache.LocalCache$LocalManualCache$1.load(LocalCache.java:4864) ~[paimon-flink-1.17-1.1-SNAPSHOT.jar:1.1-SNAPSHOT]
	at org.apache.paimon.shade.guava30.com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3529) ~[paimon-flink-1.17-1.1-SNAPSHOT.jar:1.1-SNAPSHOT]
	at org.apache.paimon.shade.guava30.com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2278) ~[paimon-flink-1.17-1.1-SNAPSHOT.jar:1.1-SNAPSHOT]
	at org.apache.paimon.shade.guava30.com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2155) ~[paimon-flink-1.17-1.1-SNAPSHOT.jar:1.1-SNAPSHOT]
	at org.apache.paimon.shade.guava30.com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2045) ~[paimon-flink-1.17-1.1-SNAPSHOT.jar:1.1-SNAPSHOT]
	... 35 common frames omitted
```

### What doesn't meet your expectations?

Exception

### Anything else?

_No response_

### Are you willing to submit a PR?

- [x] I'm willing to submit a PR!

--- a/paimon-common/src/main/java/org/apache/paimon/CoreOptions.java
+++ b/paimon-common/src/main/java/org/apache/paimon/CoreOptions.java
@@ -1827,7 +1827,7 @@ public class CoreOptions implements Serializable {
         if (keyString == null) {
             return Collections.emptyList();
         }
-        return Arrays.asList(keyString.split(","));
+        return Arrays.stream(keyString.split(",")).map(String::trim).collect(Collectors.toList());
     }
 
     public boolean fieldCollectAggDistinct(String fieldName) {

